packages/jsapi-components/src/spectrum/utils/usePickerProps.ts

```diff
@@ -77,6 +77,9 @@ export function usePickerProps<TProps>({
     [tableSource]
   );
 
+  useEffect(() => () => tableSource?.close(), [tableSource]);
```

Comment:
I would recommend using `useTableClose` hook here. It's mostly the same except it checks if table is already closed before closing it. IIR, this avoids a console warning.

Reply:
Oh actually, this is probably not a good thing to do. The tableSource is passed in and presumably managed from the outside. I don't think we want the hook to close the source.

Reply:
I moved it up to the components. Probably wouldn't have been a big deal since the hook is only used by these components, but makes sense to close it in the owner instead